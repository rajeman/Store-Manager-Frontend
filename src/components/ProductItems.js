import React from 'react';
import { connect } from 'react-redux';
import Product from './Product'
import { fetchProducts } from '../actions/products';

class ProductItems extends React.Component {
    componentDidMount() {
        this.props.dispatch(fetchProducts())
    }
    render() {
        const { products } = this.props.products;
        return (
            <div className="container">
                <div className="wrapper">
                    <div className="items-box">
                        {products.map((product) => {
                            return (
                                <Product key={product.product_id} product = {product}/>
                            );
                        })}
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => state;
export default connect(mapStateToProps)(ProductItems);
