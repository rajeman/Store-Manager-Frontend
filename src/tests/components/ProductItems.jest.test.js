import React from 'react';
import { shallow } from 'enzyme';
import "../setupTests"
import ProductItems from '../../components/ProductItems';

test('should render ProductItems correctly', () => {
    const wrapper = shallow(<ProductItems />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ProductItems with product data', () => {
    const wrapper = shallow(<ProductItems product = {[{id:1}, {id: 2}]}/>);
    expect(wrapper).toMatchSnapshot();
})
