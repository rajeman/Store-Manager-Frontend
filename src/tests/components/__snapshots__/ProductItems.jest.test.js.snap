// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProductItems correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`should render ProductItems provided 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`should render ProductItems with product data 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
